src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -839,6 +841,41 @@ class spell_warr_sweeping_strikes : public AuraScript
     Unit* _procTarget = nullptr;
 };
 
+// 394329 - Titanic Rage (attached to 385059 - Odyn's Fury)
+class spell_warr_titanic_rage : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_WARRIOR_TITANIC_RAGE, SPELL_WARRIOR_WHIRLWIND_CLEAVE_AURA });
+    }
+
+    bool Load() override
+    {
+        return GetCaster()->HasAura(SPELL_WARRIOR_TITANIC_RAGE);
+    }
+
+    void HandleCast() const
+    {
+        Unit* caster = GetCaster();
+        caster->CastSpell(nullptr, SPELL_WARRIOR_ENRAGE, CastSpellExtraArgsInit{
+            .TriggerFlags = TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR,
+            .TriggeringSpell = GetSpell()
+        });
+
+        SpellInfo const* spellInfo = sSpellMgr->AssertSpellInfo(SPELL_WARRIOR_WHIRLWIND_CLEAVE_AURA, GetCastDifficulty());
+        caster->CastSpell(nullptr, SPELL_WARRIOR_WHIRLWIND_CLEAVE_AURA, CastSpellExtraArgsInit{
+            .TriggerFlags = TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR,
+            .TriggeringSpell = GetSpell(),
+            .SpellValueOverrides = {{ SPELLVALUE_AURA_STACK, static_cast<int32>(spellInfo->StackAmount) }}
```

Comment:
This is wrong, it is affected by Meat Cleaver

Is this 1 cast in sniff or multiple (as many as you get stacks?)

Reply:
1 cast, but I overlooked that the original cast id is empty tho

Reply:
I think I addressed it correctly now